api/types/applications.go

```diff
@@ -0,0 +1,48 @@
+package types
+
+// CAT
+type CloudApplicationTemplate struct {
+	ID           string  `json:"id" header:"ID"`
+	Name         string  `json:"name" header:"NAME"`
+	Version      string  `json:"version" header:"VERSION"`
+	Global       bool    `json:"global" header:"GLOBAL"`
+	UploadURL    string  `json:"upload_url,omitempty" header:"UPLOAD_URL" show:"noshow,nolist"`
+	VendorID     string  `json:"vendor_id" header:"VENDOR_ID"`
+	Inputs       []Input `json:"inputs" header:"INPUTS" show:"nolist"`
+	IsMock       bool    `json:"is_mock" header:"IS_MOCK" show:"nolist"`
+	ResourceType string  `json:"resource_type" header:"RESOURCE_TYPE" show:"nolist"`
+	//LabelableFields
+}
+
+type Input struct {
+	Name         string                 `json:"name" header:"NAME"`
+	Description  string                 `json:"description" header:"DESCRIPTION"`
+	Type         string                 `json:"type" header:"TYPE"`
+	Required     bool                   `json:"required" header:"REQUIRED"`
+	Dependencies map[string]interface{} `json:"dependencies" header:"DEPENDENCIES"`
+}
+
+// CAD
+type CloudApplicationDeployment struct {
+	ID           string `json:"id" header:"ID"`
+	Name         string `json:"name" header:"NAME"`
+	Namespace    string `json:"namespace" header:"NAMESPACE" show:"nolist"`
+	Value        string `json:"value" header:"VALUE"`
+	CatID        string `json:"cat_id" header:"CAT_ID" show:"nolist"`
+	ResourceType string `json:"resource_type" header:"RESOURCE_TYPE" show:"nolist"`
+}
+
+// tosca task
+type DeploymentTask struct {
```

Comment:
You shoud probably call this CloudApplicationDeploymentTask or CADeploymentTask, to make it clearer its relationship to CloudApplicationDeployment